pkg/kubelet/lifecycle/handlers.go

```diff
@@ -128,8 +156,23 @@ func (hr *handlerRunner) runHTTPHandler(pod *v1.Pod, container *v1.Container, ha
 			return "", err
 		}
 	}
-	url := fmt.Sprintf("http://%s/%s", net.JoinHostPort(host, strconv.Itoa(port)), handler.HTTPGet.Path)
```

Comment:
Prior to this change, lifecycle handlers incorrectly configured with `scheme: HTTPS` actually run against http endpoints successfully.

This PR flips that behavior, meaning that on upgrade, those previous-working-but-incorrectly-specified handlers will break. I'd recommend guarding this change with a feature gate so deployers can sweep for existing incorrect lifecycle handlers independent of upgrade. See a similar escape hatch for exec probe timeouts in https://github.com/kubernetes/kubernetes/pull/94115

Reply:
Thanks, @liggitt. Is there documentation on adding features, or should I just try to go off of how it's done in #94115?

Reply:
I think following the #94115 example is reasonable, would like sig-node folks to weigh in (cc @ehashman, @SergeyKanzhelev):
* feature gate in kube_features.go like `LifecycleHandlerHTTPS` defaulting to true
* check of that feature gate in the spot where https is now honored in this PR
* if the gate is false, treat https lifecycle probes as http as was done in previous versions

Reply:
Related: I just noticed there doesn't appear to be any validation on the the `Host` parameter, so technically someone could have shoved the full URL (except scheme) into the host field, and that would have worked prior to this change... not sure if that's something we care about...

Reply:
The fact that it _would_ have worked strikes me as a bug, but that's also why there's a feature gate for these changes so that users that are relying on that broken behavior can opt-out for a release.